keys/keys.go

```diff
@@ -63,7 +63,11 @@ func MakeNamespaceMetadataKey(namespace string) proto.Key {
 
 // MakeTableMetadataKey returns the key for the table in namespaceID.
 func MakeTableMetadataKey(namespaceID uint32, tableName string) proto.Key {
-	return MakeKey(TableMetadataPrefix, encoding.EncodeUvarint(nil, uint64(namespaceID)), proto.Key(tableName))
+	k := make([]byte, 0, len(TableMetadataPrefix)+10+len(tableName))
+	k = append(k, TableMetadataPrefix...)
+	k = encoding.EncodeUvarint(k, uint64(namespaceID))
+	k = append(k, []byte(tableName)...)
```

Comment:
You shouldn't need the `[]byte` cast around `tableName`: http://play.golang.org/p/Wc6l0BFBtr

FYI, casts from `[]byte` to `string` and vice-versa require an allocation.